### PR TITLE
feat: fix function lookup in coverage tool

### PR DESCRIPTION
**Pull request by srikrishnak**

### CODE REVIEW

File: tests/coverage/extensions.py [hunk 1]
```diff
@@ -247,13 +248,15 @@ def __init__(
 
     def add_functions(self, functions, func_type):
         for func in functions.values():
-            self.extensions.add(func["uri"])
+            index = func['uri'].rfind('/extensions')
```
Comment (scgkiran) [inline tests/coverage/extensions.py]: fix should not be here. while adding to the map add it correctly, also don't do find of extensions.

File: tests/test_extensions.py [hunk 2]
```diff
@@ -58,3 +60,26 @@ def test_build_type_to_short_type():
...
+        ("add", ["dec", "dec"], "dec", "/extensions/functions_arithmetic_decimal.yaml", False),
+        ("bitwise_xor", ["dec", "dec"], "dec", "/extensions/functions_arithmetic_decimal.yaml", False),
+        # negative case, lt for i8 with wrong uri
+        ("lt", ["i8", "i8"], "bool", "/extensions/functions_datetime.yaml", True),
+        ("add", ["i8", "i8"], "i8", "/extensions/functions_arithmetic_decimal.yaml", True),
+        ("add", ["dec", "dec"], "dec", "/extensions/functions_arithmetic.yaml", True),
+        ("max", ["dec", "dec"], "dec", "/extensions/functions_arithmetic.yaml", True),
+    ]
+)
+def test_uri_match_in_get_function(func_name, func_args, func_ret, func_uri, expected_failure):
```
Comment (scgkiran) [inline tests/test_extensions.py]: wrong file for the test. it should be inside coverage folder.

File: tests/coverage/extensions.py [hunk 3]
```diff
@@ -247,13 +249,13 @@ def __init__(
...
         for func in functions.values():
-            self.extensions.add(func["uri"])
+            self.extensions.add(func['uri'])
             f_name = func["name"]
             fun_arr = self.registry.get(f_name, [])
             for overload in func["overloads"]:
                 function = FunctionVariant(
                     func["name"],
-                    func["uri"],
+                    func['uri'],
```
Comment (scgkiran) [inline tests/coverage/extensions.py]: why single quotes?

File: tests/coverage/extensions.py [hunk 4]
```diff
@@ -247,13 +249,13 @@ def __init__(
 
     def add_functions(self, functions, func_type):
         for func in functions.values():
-            self.extensions.add(func["uri"])
+            self.extensions.add(func['uri'])
```
Comment (scgkiran) [inline tests/coverage/extensions.py]: why single quotes?

File: tests/coverage/extensions.py [hunk 5]
```diff
@@ -145,43 +145,45 @@ def add_functions_to_map(func_list, function_map, suffix, extension):
...
+        extensions = dict(sorted(extensions.items()))
 
         scalar_functions = {}
         aggregate_functions = {}
         window_functions = {}
         dependencies = {}
         # convert yaml file to a python dictionary
-        for extension in extensions:
+        for extension, uri in extensions.items():
+            extension = extension
```
Comment (scgkiran) [inline tests/coverage/extensions.py]: why this?

File: tests/coverage/extensions.py [hunk 6]
```diff
@@ -145,43 +145,45 @@ def add_functions_to_map(func_list, function_map, suffix, extension):
...
-        extensions.sort()
+        extensions = dict(sorted(extensions.items()))
 
         scalar_functions = {}
         aggregate_functions = {}
         window_functions = {}
         dependencies = {}
         # convert yaml file to a python dictionary
-        for extension in extensions:
+        for extension, uri in extensions.items():
```
Comment (scgkiran) [inline tests/coverage/extensions.py]: why dict?

File: tests/coverage/extensions.py (hunk 4 above)
Comment (srikrishnak) [reply tests/coverage/extensions.py]: oops! golang took me over.

File: tests/coverage/extensions.py (hunk 6 above)
Comment (srikrishnak) [reply tests/coverage/extensions.py]: thought it would be easier to understand. removed as it is unnecessary.

**Review by scgkiran — COMMENTED** [review]
lgtm!

**Pull request closed without merging**